src/fastoad/notebooks/02_explanation/04_OpenMDAO.ipynb

```diff
@@ -0,0 +1,892 @@
+{
```

Comment:
<p>At first, I wondered if it was worth bothering users with IndepVarComp (that's old-school since the auto-ivc feature :D ). <code>problem.set_val()</code> does perfectly the job, after all.</p><p>Then I realized that these inputs are used several times, and then it may avoid some repetition (and that would be worth mentioning IMHO).</p><p> </p><p>In a larger point of view, I like how you show the usage of N2 charts to spot missing entries, and I am tempted to go a bit further. My idea would be:</p><ul><li>Setting up the problem without IVC and plotting the N2 chart. It will show all the variables from auto-ivc, meaning their default value will be used.</li><li>If we keep the empty component for OWE computation, we may point out that OWE is not expected to be in the auto-ivc. In that case, I even suggest to not have an empty OWE computation and to have a mass group that does not even contain the OWE component (maybe just a comment to say it should be added). Then the user would be encouraged to fix that (and <strong>to restart the kernel once it is done</strong>). And to allow the notebook to run correctly anyway, we may create another problem containing a group from <code>correction/</code> that would have a correct OWE computation... and then plot again the N2 chart.</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fast-aircraft-design/FAST-OAD/pull/426/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>